.github/workflows/pytest.yml

```diff
@@ -1,5 +1,5 @@
 name: pytest
-on: [push, pull_request]
+on: [push]
```

Comment:
it seemed mandatory to execute the tests. At least in #241 and #238 no tests and code analysis steps were executed without the pull_request trigger.

Reply:
I know but for some reason, it's different on this branch. I guess it was some weird behavior on the other PR.

Reply:
do we add this back in before merging?